app/javascript/components/tables/hooks/useTableSortingAndFiltering.js

```diff
@@ -0,0 +1,234 @@
+   * @param {Object} e change event from the search input field
+   */
+  const handleSearch = (e) => {
+    const filter = e.target.value;
+    setSearchFilter(filter);
+    filterRows(filter, tagFilter);
+  };
+
+  /**
+   * Handle tag filter changes
+   * @param {Object} e change event from the tag filter drop-down
+   */
+  const handleTagFilter = (e) => {
+    const filter = e.target.value;
+    setTagFilter(filter);
+    filterRows(searchFilter, filter);
+  };
+
+  /**
+   * Sort rows by a given column key and direction (asc/desc)
+   * @param {Array} rows
+   * @param {String} columnKey column name to be used for sorting
+   * @param {String} direction sorting direction (asc/desc)
+   * @param {String} dataType data type of the column (string, number, date)
+   * @returns {Array}
+   */
+  const sortRows = (rows, columnKey, direction = 'asc', dataType = 'string') => {
+    let sorted;
+
+    switch (dataType) {
+      case 'date':
+        // Sort by date values from the parsed timestamps
+        sorted = rows.sort((a, b) => {
+          return a[columnKey] - b[columnKey];
+        });
+        break;
+
+      case 'number':
+        // Sort by numeric values
+        sorted = rows.sort((a, b) => {
+          const numA = parseFloat(a[columnKey]) || 0;
+          const numB = parseFloat(b[columnKey]) || 0;
+          return numA - numB;
+        });
+        break;
+
+      case 'string':
+      default:
+        // Case-insensitive sort with stable fallback
+        sorted = rows.sort((a, b) => {
+          const aVal = (a[columnKey] || '').toString().toLowerCase();
+          const bVal = (b[columnKey] || '').toString().toLowerCase();
+
+          // Primary sort: case-insensitive comparison
+          if (aVal < bVal) return -1;
+          if (aVal > bVal) return 1;
+
+          // Secondary sort: original case-sensitive value for stable sort
+          const aOriginal = (a[columnKey] || '').toString();
+          const bOriginal = (b[columnKey] || '').toString();
+          if (aOriginal < bOriginal) return -1;
+          if (aOriginal > bOriginal) return 1;
+          return 0;
+        });
+        break;
+    }
+
+    return direction === 'asc' ? sorted : sorted.reverse();
+  };
+
+  /**
+   * Handl sorting when a column header is clicked
+   * @param {Number} columnIndex column index of the sorting column
+   * @returns 
+   */
+  const handleSort = (columnIndex) => {
+    if (!columns[columnIndex].sortable) return;
+
+    const column = columns[columnIndex];
+    const columnKey = column.key;
+    const dataType = column.dataType || 'string';
+
+    let sortDirection;
+    if (sorting.column === columnIndex) {
+      // Toggle direction if the same column is clicked
+      sortDirection = sorting.direction === 'asc' ? 'desc' : 'asc';
+    } else {
+      // Choose appropriate initial direction based on data type
+      if (dataType === 'date') {
+        // Date columns: starts with oldest first
+        sortDirection = 'asc';
```

Comment:
Did our previous implementation(s) default to date ascending? I feel like in most cases a user would be most interested in recent activity, so default date sort should be descending.

Reply:
Yes, we were sorting dates in the ascending order in the initial sort. The only place we were sorting in the descending order was in the encoding jobs table.
I agree it would make more sense to sort in descending order when it comes to date fields. I will make that change 👍